drv/stm32h7-spi-rot-server/src/main.rs

```diff
@@ -0,0 +1,274 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#![no_std]
+#![no_main]
+
+use idol_runtime::{ClientError, Leased, RequestError, R, W};
+// use idol_runtime::{NotificationHandler, ClientError, Leased, RequestError, R, W};
+use drv_spi_api::{CsState, Spi, SpiError};
+use drv_spi_msg::*;
+use drv_stm32xx_sys_api as sys_api;
+use userlib::*;
+
+task_slot!(SPI, spi_driver);
+task_slot!(SYS, sys);
+
+use ringbuf::*;
+
+const SPI_TO_ROT_DEVICE: u8 = 0;
+
+// On Gemini, the STM32H753 is in a LQFP176 package with ROT_IRQ on pin2/PE3
```

Comment:
Can you add a config block here that will fail to compile on non-Gemini boards, so no one tries to run this on Gimlet by accident?